src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/PropertySymbolReferenceFinder.cs

```diff
@@ -183,43 +188,21 @@ private static bool IsForEachProperty(IPropertySymbol symbol)
-            FindReferencesSearchOptions options, CancellationToken cancellationToken)
+        private static IEnumerable<SyntaxNode> GetIndexerReferenceExpressions(SyntaxNode syntaxRoot, ISyntaxFactsService syntaxFacts)
         {
-            if (options.AssociatePropertyReferencesWithSpecificAccessor)
-            {
-                // can't find indexer get/set accessors in a cref.
-                return ImmutableArray<FinderLocation>.Empty;
-            }
-
-            var symbolsMatch = GetStandardSymbolsNodeMatchFunction(symbol, document.Project.Solution, cancellationToken);
-
-            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            var semanticFacts = document.GetLanguageService<ISemanticFactsService>();
-
-            // Now that we have Doc Comments in place, We are searching for References in the Trivia as well by setting descendIntoTrivia: true
-            var indexerMemberCrefs = syntaxRoot.DescendantNodes(descendIntoTrivia: true)
-                                               .Where(syntaxFacts.IsIndexerMemberCRef);
-
-            var locations = ArrayBuilder<FinderLocation>.GetInstance();
+            var elementAccessExpressions = syntaxRoot.DescendantNodes().Where(syntaxFacts.IsElementAccessExpression);
 
-            foreach (var node in indexerMemberCrefs)
+            var conditionalAccessExpressions = syntaxRoot.DescendantNodes().Where(syntaxFacts.IsConditionalAccessExpression).Select(node =>
             {
-                cancellationToken.ThrowIfCancellationRequested();
+                syntaxFacts.GetPartsOfConditionalAccessExpression(node, out var conditionalExpression, out var whenNotNull);
+                return whenNotNull;
+            });
 
-                var (matched, reason) = symbolsMatch(node, semanticModel);
-                if (matched)
-                {
-                    var location = node.SyntaxTree.GetLocation(new TextSpan(node.SpanStart, 0));
-                    var symbolUsageInfo = GetSymbolUsageInfo(node, semanticModel, syntaxFacts, semanticFacts, cancellationToken);
-                    locations.Add(new FinderLocation(
-                        node, new ReferenceLocation(document, null, location, isImplicit: false, symbolUsageInfo, GetAdditionalFindUsagesProperties(node, semanticModel, syntaxFacts), candidateReason: reason)));
-                }
-            }
+            // Now that we have Doc Comments in place, We are searching for References in the Trivia as well by setting descendIntoTrivia: true
+            var indexerMemberCrefs = syntaxRoot.DescendantNodes(descendIntoTrivia: true).Where(syntaxFacts.IsIndexerMemberCRef);
 
-            return locations.ToImmutableAndFree();
+            return elementAccessExpressions.Concat(conditionalAccessExpressions)
+                                           .Concat(indexerMemberCrefs);
         }
```

Comment:
i think it would be nicer to just have: `syntaxRoot.DescendantNodes(descendIntoTrivia: true).Where(... IsElementAccess || IsIndexerCref || IsConditionalAccessExpression)`.  Then in the binding code you distinguish those three cases.